backend_plugin/src/main/evaluators/BlockCommentEvaluator.java

```diff
@@ -3,114 +3,118 @@
 import org.eclipse.jface.text.BadLocationException;
 import org.eclipse.jface.text.DocumentEvent;
 import org.eclipse.jface.text.IDocument;
-import org.eclipse.jface.text.IRegion;
 
-/**
- * Evaluates DocumentEvent changes to determine if the user is commenting out multiple sequential lines of code. If so,
- * then the user should be notified of the block comment feature in Eclipse.
- */
 public class BlockCommentEvaluator extends FeatureEvaluator {
+
+	private int lastCommentedLine;
+	private long lastCommentedLineTimeStamp;
 		
-	private boolean firstDoubleBackSlashDetected;
-	private IRegion prevRegion;
-
-	// position of the previous document change
-	private int prevOffset;
-	// text of the previous document change
-	private String prevInsert;
-
-	// position of the current document change
-	private int currentOffset;
-	// text of the current document change
-	private String currentInsert;
-
-	private static final String SINGLE_SLASH = "/";
-	private static final String DOUBLE_SLASH = "//";
-	
 	/**
 	 * Default constructor
 	 */
 	public BlockCommentEvaluator() {
 		this.featureID = "blockCommentSuggestion";
-		firstDoubleBackSlashDetected = false;
+		this.lastCommentedLine = -2;
```

Comment:
I chose -2 instead of -1 for the case where the user has freshly opened a document and has commented out the very first line (Since the first line is line 0, it would falsely detect an adjacent line, -1 in this case, even though its not a real line). Not exactly a likely case, but I figured I might as well cover for it